### PR TITLE
[NanoAOD, JMENano] Updates for Nano and JMENano

### PR DESCRIPTION
**Pull request by nurfikri89**
#### PR description:

This PR aims to add changes related to NanoAOD, JMENanoAOD and ensuring consistency of PuppiProducer instances when processing MiniAODs for production and analysis. The following changes are made:

**NanoAOD**
- Add missing `RUN3WINTER22PUPPIrunsBCDEprompt` JetID version for jets.
- Fix more typo in `nanoAOD_addDeepInfoAK4()`.
- Add (puppi-weighted) charged and neutral multiplicities for jets.
- Add HF energy fractions for AK4 jets.
- Add jet area for subjets. Improve documentation of collection.

**JMENanoAOD**
- Add Run-3 JetIDs.
- Remove AK8CHS jet collection.
- Reduce content for AK4CHS collection by removing PUID/QGL tagger inputs and most flavour-tagging discriminants.
- Increase pT thresholds for AK4CHS, AK4GEN, AK8GEN.
- Rename some branches.

**PuppiProducer consistency**
- Improve `setupPuppiForPackedPF()` in `jetTools.py` by adding `packedpuppiNoLep` instance.
- In `runMETCorrectionsAndUncertainties.py`, make it optional to provide a different `PuppiProducer` module label rather than stick to a hard-coded label (`puppiNoLep`). Set the previously hard-coded label as the default option.

#### PR validation:
- passes the standard runTheMatrix test: `runTheMatrix.py -l limited -i all --ibeos`
- passes the following NanoAOD workflows: `runTheMatrix.py -i all --ibeos -l 2500.0,2500.001,2500.002,2500.01,2500.011,2500.012,2500.1,2500.2,2500.21,2500.3,2500.4`

#### If this PR will be backported please specify to which release cycle the backport is meant for:
To be backported to earlier 13_Y releases for Run-3 JMENano production.

### CODE REVIEW

File: PhysicsTools/NanoAOD/python/jetsAK4_CHS_cff.py [hunk 1]
```diff
@@ -171,11 +171,15 @@
         hfcentralEtaStripSize = Var("userInt('hfJetShowerShape:centralEtaStripSize')", int, doc="eta size of the central tower strip in HF (noise discriminating variable) "),
         hfadjacentEtaStripsSize = Var("userInt('hfJetShowerShape:adjacentEtaStripsSize')", int, doc="eta size of the strips next to the central tower strip in HF (noise discriminating variable) "),
         nConstituents = Var("numberOfDaughters()","uint8",doc="Number of particles in the jet"),
+        chMultiplicity = Var("chargedMultiplicity()","uint8",doc="Number of charged particles in the jet"),
```
Comment (simonepigazzini) [inline PhysicsTools/NanoAOD/python/jetsAK4_CHS_cff.py:174]: are these changes intended? I didn't expect to see changes to the nano content in this PR

**Comment by nurfikri89** [comment]
I am closing this PR and will submit another one using the latest IB release to include the changes in PR #42670.
